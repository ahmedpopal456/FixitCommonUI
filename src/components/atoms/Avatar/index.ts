import { AvatarBadge } from './Avatar';
export { AvatarBadge};