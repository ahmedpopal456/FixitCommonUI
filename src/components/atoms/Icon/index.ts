import { Icon } from './Icon';
import { IconLibraries } from './IconLibraries';

export { Icon, IconLibraries };
