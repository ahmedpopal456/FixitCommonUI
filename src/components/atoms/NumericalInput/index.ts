import { NumericalInput } from './NumericalInput';

export { NumericalInput };
