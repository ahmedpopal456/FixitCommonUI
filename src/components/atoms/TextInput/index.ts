import { TextBox } from './TextInput';

export { TextBox };
