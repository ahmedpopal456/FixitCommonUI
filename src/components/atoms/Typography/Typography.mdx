---
name: Typography
menu: Atoms
---
import { Props, Playground } from "docz";
import { H1, H2, H3, H4, H5, Label, P } from "./Typography.tsx";

# Typography
The Typography components include **H1**, **H2**, **H3**, **H4**, **H5**, **P** and **Label**.

## Props
All typography elements have the same Props, but with different default values.
The default values in the folowing table are from the **P** component.
<Props of={P}></Props>

# Usage
```tsx
import { H1 } from "fixit-common-ui";

<H1>Typography</H1>
```

## Default
<Playground>
  <H1>Typography</H1>
  <H2>Typography</H2>
  <H3>Typography</H3>
  <H4>Typography</H4>
  <H5>Typography</H5>
  <Label>Typography</Label>
  <P>Typography</P>
</Playground>

## Caps
Passing th `$caps` prop will capitalize every character in the text.
<Playground>
  <H1 $caps>Typography</H1>
  <H2 $caps>Typography</H2>
  <H3 $caps>Typography</H3>
  <H4 $caps>Typography</H4>
  <H5 $caps>Typography</H5>
  <Label $caps>Typography</Label>
  <P $caps>Typography</P>
</Playground>

## Styling
If the default styling need to be changed, a fontSize, fontWeight and color prop can be passed.
If more customization is needed, you can pass a style prop.
<Playground>
  <P>Default</P>
  <P fontSize={30}>Font Size</P>
  <P fontWeight={800}>Font Weight</P>
  <P color="accent">Color Accent</P>
  <P style={{
    color: "white", 
    fontSize: 24, 
    backgroundColor: "black", 
    textAlign: "center",
    padding: 5,
    textTransform: "uppercase"}}
  >Styling</P>
</Playground>
