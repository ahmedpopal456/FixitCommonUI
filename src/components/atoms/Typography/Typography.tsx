import React from "react";
import { StyledText, StyledContainer } from "./style"

import { TypographyProps } from "./TypographyModel";

/** 
 * @param props - The H1 component props
 * @returns A react native custom H1 component
 */
export const H1: React.FC<TypographyProps> = (props): JSX.Element => {
  return renderText(props);
};

H1.defaultProps = {
  fontSize: 24,
  fontWeight: 400,
  color: "dark",
  padding: 3,
  justification: "left"
};

/** 
 * @param props - The H2 component props
 * @returns A react native custom H2 component
 */
export const H2: React.FC<TypographyProps> = (props): JSX.Element => {
  return renderText(props);
};

H2.defaultProps = {
  fontSize: 18,
  fontWeight: 700,
  color: "dark",
  padding: 3,
  justification: "left"
};

/** 
 * @param props - The H3 component props
 * @returns A react native custom H3 component
 */
export const H3: React.FC<TypographyProps> = (props): JSX.Element => {
  return renderText(props);
};

H3.defaultProps = {
  fontSize: 15,
  fontWeight: 400,
  color: "dark",
  padding: 3,
  justification: "left"
};

/** 
 * @param props - The H4 component props
 * @returns A react native custom H4 component
 */
export const H4: React.FC<TypographyProps> = (props): JSX.Element => {
  return renderText(props);
};

H4.defaultProps = {
  fontSize: 13,
  fontWeight: 400,
  color: "dark",
  padding: 3,
  justification: "left"
};

/** 
 * @param props - The H5 component props
 * @returns A react native custom H5 component
 */
export const H5: React.FC<TypographyProps> = (props): JSX.Element => {
  return renderText(props);
};

H5.defaultProps = {
  fontSize: 13,
  fontWeight: 400,
  color: "grey",
  padding: 3,
  justification: "left"
};

/** 
 * @param props - The Label component props
 * @returns A react native custom Label component
 */
export const Label: React.FC<TypographyProps> = (props): JSX.Element => {
  return renderText(props);
};

Label.defaultProps = {
  fontSize: 13,
  fontWeight: 700,
  color: "dark",
  padding: 3,
  justification: "left"
};
    
/** 
 * @param props - The P component props
 * @returns A react native custom P component
 */
export const P: React.FC<TypographyProps> = (props): JSX.Element => {
  return renderText(props);
};

P.defaultProps = {
  fontSize: 13,
  fontWeight: 400,
  color: "dark",
  padding: 8,
  justification: "left"
};
    
/** 
 * Since all the typography elements are so similar appart from the default props
 * this method renders the text.
 * 
 * @param props - The component's props
 * @returns A react native custom Typography component based on the given props
 */
const renderText = (props: TypographyProps) => {
  return (
    <StyledContainer>
      <StyledText {...props}>
        {props.children}
      </StyledText>
    </StyledContainer>
  );
}
