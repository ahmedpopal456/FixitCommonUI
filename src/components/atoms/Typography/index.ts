import {
  H1, H2, H3, H4, H5, Label, P,
} from './Typography';

export {
  H1, H2, H3, H4, H5, Label, P,
};
