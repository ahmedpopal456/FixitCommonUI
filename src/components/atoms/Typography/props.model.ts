import colors from '../../../theme/colors';

export interface Props {
    /**  To override default style */
    style?: any;
    /** Can be any color from the theme color palette */
    color?: keyof typeof colors;
    /** Size of the text */
    fontSize?: number;
    /** Weight of the text */
    fontWeight?: number;
    /** Capitalizes the characters */
    $caps?: boolean;
    /** Padding value for the text */
    padding?: number;
    /** Justification */
    justification?: "left" | "center" | "right",
    /** Used for testing */
    testID?: string;
    children: any;
  }