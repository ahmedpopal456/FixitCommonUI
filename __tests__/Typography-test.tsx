import "react-native";
  import React from "react";
  import { Label, H1, H2 } from "../src/components/atoms/Typography";
  
  // Note: test renderer must be required after react-native.
  import renderer from "react-test-renderer";
  import { render, fireEvent } from '@testing-library/react-native';
  
  it('renders correctly', () => {
      renderer.create(<Label>Test</Label>);
  });
  