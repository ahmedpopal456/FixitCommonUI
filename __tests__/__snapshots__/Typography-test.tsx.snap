// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches $caps snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <Text
    color="dark"
    fontSize={13}
    fontWeight={400}
    justification="left"
    padding={8}
    style={
      Array [
        Object {
          "color": "#1D1F2A",
          "flexDirection": "row",
          "fontFamily": "Raleway",
          "fontSize": 13,
          "fontWeight": "400",
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
          "textAlign": "left",
          "textTransform": "uppercase",
        },
      ]
    }
  >
    Button
  </Text>
</View>
`;

exports[`matches H1 snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <Text
    color="dark"
    fontSize={24}
    fontWeight={400}
    justification="left"
    padding={3}
    style={
      Array [
        Object {
          "color": "#1D1F2A",
          "flexDirection": "row",
          "fontFamily": "Raleway",
          "fontSize": 24,
          "fontWeight": "400",
          "paddingBottom": 3,
          "paddingLeft": 3,
          "paddingRight": 3,
          "paddingTop": 3,
          "textAlign": "left",
        },
      ]
    }
  >
    Button
  </Text>
</View>
`;

exports[`matches H2 snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <Text
    color="dark"
    fontSize={18}
    fontWeight={700}
    justification="left"
    padding={3}
    style={
      Array [
        Object {
          "color": "#1D1F2A",
          "flexDirection": "row",
          "fontFamily": "Raleway",
          "fontSize": 18,
          "fontWeight": "700",
          "paddingBottom": 3,
          "paddingLeft": 3,
          "paddingRight": 3,
          "paddingTop": 3,
          "textAlign": "left",
        },
      ]
    }
  >
    Button
  </Text>
</View>
`;

exports[`matches H3 snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <Text
    color="dark"
    fontSize={15}
    fontWeight={400}
    justification="left"
    padding={3}
    style={
      Array [
        Object {
          "color": "#1D1F2A",
          "flexDirection": "row",
          "fontFamily": "Raleway",
          "fontSize": 15,
          "fontWeight": "400",
          "paddingBottom": 3,
          "paddingLeft": 3,
          "paddingRight": 3,
          "paddingTop": 3,
          "textAlign": "left",
        },
      ]
    }
  >
    Button
  </Text>
</View>
`;

exports[`matches H4 snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <Text
    color="dark"
    fontSize={13}
    fontWeight={400}
    justification="left"
    padding={3}
    style={
      Array [
        Object {
          "color": "#1D1F2A",
          "flexDirection": "row",
          "fontFamily": "Raleway",
          "fontSize": 13,
          "fontWeight": "400",
          "paddingBottom": 3,
          "paddingLeft": 3,
          "paddingRight": 3,
          "paddingTop": 3,
          "textAlign": "left",
        },
      ]
    }
  >
    Button
  </Text>
</View>
`;

exports[`matches H5 snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <Text
    color="grey"
    fontSize={13}
    fontWeight={400}
    justification="left"
    padding={3}
    style={
      Array [
        Object {
          "color": "#8B8B8B",
          "flexDirection": "row",
          "fontFamily": "Raleway",
          "fontSize": 13,
          "fontWeight": "400",
          "paddingBottom": 3,
          "paddingLeft": 3,
          "paddingRight": 3,
          "paddingTop": 3,
          "textAlign": "left",
        },
      ]
    }
  >
    Button
  </Text>
</View>
`;

exports[`matches Label snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <Text
    color="dark"
    fontSize={13}
    fontWeight={700}
    justification="left"
    padding={3}
    style={
      Array [
        Object {
          "color": "#1D1F2A",
          "flexDirection": "row",
          "fontFamily": "Raleway",
          "fontSize": 13,
          "fontWeight": "700",
          "paddingBottom": 3,
          "paddingLeft": 3,
          "paddingRight": 3,
          "paddingTop": 3,
          "textAlign": "left",
        },
      ]
    }
  >
    Button
  </Text>
</View>
`;

exports[`matches P snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <Text
    color="dark"
    fontSize={13}
    fontWeight={400}
    justification="left"
    padding={8}
    style={
      Array [
        Object {
          "color": "#1D1F2A",
          "flexDirection": "row",
          "fontFamily": "Raleway",
          "fontSize": 13,
          "fontWeight": "400",
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
          "textAlign": "left",
        },
      ]
    }
  >
    Button
  </Text>
</View>
`;
